### DISCUSSION TITLE
JS Framework

### DISCUSSION THREAD

**Discussion by TimGeyssens** [TimGeyssens/Umbraco-v8-Community-Website-StarterKit#11] [General]
Wouldn't add a full blown framework but maybe use https://github.com/alpinejs/alpine/ to sprinkle some js magic where it makes sense... 

We can then have the #9 headless option to do a full blown (small) PWA

**Comment by danbramall** [comment]
I'd ask what problem is sought to be solved by introducing a dependency on Alpine JS?  Sure, for headless/PWAs a framework can be massively beneficial (but in those cases probably not Alpine!).  If it's just a 'sprinkling' of JS I'd personally prefer to see nice, modern, vanilla JS unless a framework is going to either widen adoption or ease development.  Any dependency is a filter on who will use the end-product; the more there are, the fewer people will make it through.

**Reply by TimGeyssens** [reply]
Sure, only use where it makes sense... good call!

**Reply by skttl** [reply]
AlpineJS just makes it a lot easier to connect the dots for human beings not knowing their way around the code (of course only if they know AlpineJS or similar frameworks like Vue / angularjs etc.)

Basically AlpineJS is just a way to inject JS into your markup where you need it, there is not much not-vanilla in it.

**Comment by dawoe** [comment]
No framework... the goal is to teach umbraco... you don't want to burden users by learrning a js framework... Or you could do 2 options... plain javascript and an vanilla js view.

Dave

**Reply by TimGeyssens** [reply]
ok, so plain vanilla for the main site... and then we have a #9  headless app that could use a framework...
